docs: Blog post about JDK support in the next Scala LTS

tgodzik opened the pull request:
- [ ] PR that updates https://docs.scala-lang.org/overviews/jdk-compatibility/overview.html appropriately by linking to this very blog post
- [x] make a contributors thread for discussion
- [x] update the end of the blog post to link to the new thread

SethTisue commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,82 @@
+For the compiler’s second LTS version, the Scala 3 compiler team will be
+dropping support for JDK 8 and is considering which later JDK to use.
+
+## Reasons for the change
```
The difference between "Reasons for the change" and "Advantages" seems unclear to me.

I can see that you're trying to tell it as a story: first we hit this `Unsafe` issue, then we started thinking about it more broadly. But I don't think that story is conveyed clearly. "Reasons for the change" suggests that the section will contain the reasons. But it doesn't, many of the reasons are in the next section.

Maybe it's as simple as changing "Reasons for the change" to "Immediate motivation" and "Advantages" to "Longer-term motivation"? Something like that?

tgodzik replied:
Good idea, I did `long term motivation` instead though, what do you think?

SethTisue replied:
LGTM

SethTisue commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,82 @@
+by: Tomasz Godzik
+---
+
+**TLDR;**
```
in the TLDR section I would suggest omitting any mention of `Unsafe`. That isn't the lede; it's a technical detail. The main things the reader wants to know are: we're dropping 8; we might drop 11 too; and when this is expected to happen.

There is already material about `Unsafe` just a few lines down, anyway.

Gedochao replied:
I'd perhaps mention it in broad strokes and leave a link to the tied issue, for whoever is interested?

tgodzik replied:
Removed it and simplified the sentence

SethTisue commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,82 @@
+Scala ecosystem to start using new features brought in by JDK 9+. This might
+include a number of JEPs, which will be investigated by the team the coming
+year. To check the current status take a look at the issues marked with the
+[JEP label](https://github.com/scala/scala3/issues?q=is%3Aissue+is%3Aopen+label%3Acompat%3Ajava%3Ajep).
```
in addition to providing the link, I would suggest briefly highlighting 2 or 3 of the most important ones. Just giving the link has a sort of "you figure it out, we can't be bothered to summarize it for you" vibe

tgodzik replied:
I added some, but I am not sure how well I understand them

SethTisue replied:
heh, well me either, maybe @sjrd would like to take a look?

sjrd replied:
I would remove JEP 334. I don't think this is useful to us.

We could mention JEP 409 (sealed classes). Similarly to nest-based access control, we could produce bytecode with restrictions that better match Scala's language-level restrictions. Not a big deal, but there is some use case for it.

Looking further into the future, the addition of [`SwitchBootstraps`](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/runtime/SwitchBootstraps.html) in JDK 21 *might* allow us to encode long pattern matches more efficiently for the JVM.

tgodzik replied:
Thanks, I will switch 334 to 409. Should we open an issue for SwitchBootstraps? Was that part of a JEP or added independently?

tgodzik replied:
Ok, I found that it's 441 (at least it looks like part of it) and added it to the list

He-Pin replied:
I  told @liach about this, and the jvm team here, they suggested to Java 21 as a new foundation,  the Java 21 will be the new Java 8 in next N years, and Java 25 the next LTS have a fix for Virtual Threads, which will accelerate the adoption too.

At work,  mostly we are just migrating our system from Java 8 to Java 21, and the jvm team are putting more energy in Java 21 and Java 11.

Move to Java 21 help Scala on pattern matching with records too, and the language can leverage VT, typeswitch , zgc things which are added in Java 21.

tgodzik replied:
Makes sense! We can discuss it further once the blogpost is out and we have a contributors thread. This blogpost is just to announce that we are moving towards a newer jdk

He-Pin replied:
https://javaalmanac.io for api comparing.

liach replied:
For bumping, I think the most significant advantages in the JVM may be:
1. JEP 181 Nestmates, to eliminate bridge methods 
2. JEP 309 Condy, more simple representation of bootstrap method/indy args
3. JEP 371 Hidden classes, useful to enclose methods generated to implement indy call sites (such as dynamic implementation for pattern matching)

tgodzik replied:
@liach these are really examples, there are no actual decisions yet in this document. It's worth to make sure those issues exist in Scala 3 repo

SethTisue commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,82 @@
+Another advantage would be reducing the maintenance burden on tooling and
+library authors that currently have to take into account a large number of
+different versions to test and make sure that their code is performant on all of
+them.
```
```suggestion
them. Thus, the benefits of this change will be felt ecosystem-wide.
```

SethTisue commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,82 @@
+them.
+
+If we take a look at the timeline we'll see that JDK 8 was first published on
+18th March 2014, which is already over 10 years ago. Technology and especially
```
I suggest including some mention of whether JDK 8 is formally EOL yet. (And if so, how long ago; and if not, when is it expected.)

tgodzik replied:
It depends on the distribution, which is why I mentioned is as `A lot of the existing distributions are
already stopping or will soon stop updating JDK 8 with security and other fixes,` Should I explicitly write end of life instead?

SethTisue replied:
Actually saying "end of life" would have a certain persuasive force, but now that I look at https://www.oracle.com/java/technologies/java-se-support-roadmap.html , I see that Oracle hasn't actually cut JDK 8 off entirely yet, it just says: 

> Oracle intends to provide at least 18 months notice on this page and other communication channels if an end of availability date is set.

So I guess the text is fine as is.

SethTisue commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,82 @@
+which might directly impact issues within your business applications.
+
+Some larger projects within JVM and Scala ecosystems have already dropped JDK 8
+support. And while it's always possible to not update your libraries and Scala
```
Do we know of any especially prominent examples? (Akka? Spark? Play? famous Java things like, idk, Spring or whatever?) I think a few examples would do a lot to convince the reader that the world is already moving on and we're just joining the stampede.

SethTisue replied:
Also do we have any usage statistics on Java 8? Like from the last survey we did, for example? If the Java 8 numbers are fairly low, that's ammunition in favor of the change.

tgodzik replied:
> Do we know of any especially prominent examples? (Akka? Spark? Play? famous Java things like, idk, Spring or whatever?) I think a few examples would do a lot to convince the reader that the world is already moving on and we're just joining the stampede.

From what I remember there are some like jsoniter, mdoc, lsp4j, flywaydb, eclipse LSP server, Metals, Bloop. Obviously, some of those are ours, so it might not be that big of a wave. That's why I wrote 'some' :sweat_smile: 

> Also do we have any usage statistics on Java 8? Like from the last survey we did, for example? If the Java 8 numbers are fairly low, that's ammunition in favor of the change.

We have https://scalasurvey2023.virtuslab.com/ last I think and it was at 25%. I would think that's mostly Scala 2 projects thoguh

SethTisue replied:
The 25% number might be worth including, especially if we include a sentence like "we assume that most of those are using Scala 2 rather than 3", which I agree is a pretty safe assumption.

I checked and Spark 3.5.4 still supports 8, but Spark 4 will require 17+, as per https://github.com/apache/spark/pull/43005#issuecomment-1731344040

Play dropped 8 a while back: https://github.com/playframework/playframework/issues/11078

Akka has dropped 8: https://github.com/akka/akka/pull/32127

Spring 6 dropped 8 and 11 all the way back in 2022, as per https://spring.io/blog/2022/11/16/spring-framework-6-0-goes-ga

I think this kind of research is worth doing, as I believe it adds substantial force to the case for dropping it. "Lots of people are dropping 8" is probably the single most powerful thing we can say, IMO — if we can back it up.

SethTisue replied:
let's try a little crowdsourcing? I put a request on the Scala Discord for additions to this list

tgodzik replied:
Thanks! That is already a great list.

mkurz replied:
Play will drop Java 11 with next release and will require Java 17

reardonj replied:
From Java land, KIE (Drools and related tools) is Java 17+ in the latest release.

BalmungSan replied:
From **Java** land:

- **Logback** `1.5.x` series _(current main development branch)_ requires **Java** `11+`: https://logback.qos.ch/dependencies.html
- Not that popular, but the **Neo4j** **Java** driver `5.x` series _(current main development branch)_ require **Java** `17+`: https://github.com/neo4j/neo4j-java-driver?tab=readme-ov-file#supported-driver-series - Thus, does the **Scala** wapper, **neotypes**: https://neotypes.github.io/neotypes/

From **Scala** land:

- While **fs2** supports **Java** `8` - It has a couple of APIs that are only available when using **Java** `9+`, for example the interop with `Flow`.
- Additionally, AFAIK, one of the main reasons why **typelevel** projects continue to support **Java** `8` is because **Scala** itself does. Meaning, this change may impulse them to drop support as well. Although, maybe it would be better to ask Arman or Daniel directly to confirm.

mkurz replied:
HikariCP requires Java 11 since its 5.0 release in 2021: https://github.com/brettwooldridge/HikariCP#requirements

ghostdogpr replied:
zio dropped it a year ago: https://github.com/zio/zio/pull/8434

SethTisue replied:
on Discord, Li Haoyi said:

> com-lihaoyi land has dropped Java 8 across the board and bumped the minimum version to 11

tgodzik replied:
Thanks everyone! I added the biggers projects to the list. Let me know if you feel we should include even more. Looks like a considerable list already, so maybe that's enough?

He-Pin replied:
Pekko will drop Java 8 support after 1.2.x, Java 8 is a pain to maintain.

SethTisue commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,82 @@
+use libraries compiled with earlier JDK 8 compatible Scala 3 versions in the
+versions with a new implementation of lazy values, which is not reliant on
+sun.misc.Unsafe.
+
```
curious if there's a ticket or discussion or something on this I could follow; other readers might be curious as well. how confident are we that this can even be accomplished? 

Oh, I now see that there is a link below o ticket 9103, but does that ticket cover both:
• the new implementation
• how to be compatible with the old implementation?

tgodzik replied:
I think that needs to cover both. We can't switch to new implementation without supporting the old one.

SethTisue commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,82 @@
+
+The current plan is to drop support of JDK 8 in one of the future minor releases
+of Scala 3 and in the next LTS. The JDK will be either 11 or 17 depending on the
+community feedback and our investigations. The current estimate for the next LTS
```
Where is the community invited to leave this feedback? Is there a forum thread? (A common pattern is to open a forum thread just before the blog post is published, then include the link in the post.)

tgodzik replied:
That's exactly my plan. We are still far from publishing as I see :sweat_smile:

SethTisue commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,82 @@
+LTS under 3.3.x will be supported for at least another year after the release of
+the next LTS version, which should give you plenty of time to migrate.
+
+If at any point it turns out it's not possible for you to switch, be sure to
```
The wording here feels strange to me. "If at any point" is too late. Once this decision is made it will be irreversible. We need the feedback now, I think?

I think we should be clear and honest about whether we are _only_ asking the community "11 vs 17", or whether we are also inviting feedback on whether this is happening in the next LTS at all.

tgodzik replied:
I changed it so that it's clearer. We can try to solve people's issues with migration if they are technical.

SethTisue commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,82 @@
+If at any point it turns out it's not possible for you to switch, be sure to
+send us your feedback.
+
+## Summary
```
```suggestion
## Discussions and further reading
```

SethTisue reviewed (commented):
In this round, I've tried to only give substantive feedback and not nitpick the grammar and such.

Before this is published, I would like to have a second opportunity to polish the prose and the flow.

SethTisue commented:
I suggest also having a second PR that also updates https://docs.scala-lang.org/overviews/jdk-compatibility/overview.html appropriately — basically by linking to this very blog post.

He-Pin commented:
I think Java 21 is better

tgodzik commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,82 @@
+## Advantages
+
+Switching to a newer version of the JDK would allow both the compiler and the
+Scala ecosystem to start using new features brought in by JDK 9+. This might
```
changed

SethTisue commented:
someone already shared this on the Scala Reddit, though they withdrew it when I objected. I've marked the post as 'draft" in the hopes of discouraging wide sharing.

tgodzik commented:
> someone already shared this on the Scala Reddit, though they withdrew it when I objected. I've marked the post as 'draft" in the hopes of discouraging wide sharing.

It's not really a secret, but it's weird to share before being published.

JD557 commented:
There's one thing that bugs me about Scala dropping Java 8: It is now going to be more troublesome to distribute Scala applications to desktop users.

For example, I believe that the traditional way would be to distribute the `.jar` and mention "This software requires Java installed".

This is an issue as, when a user searches for "Download Java" in Google, it gets pointed to https://java.com, which provides only Java 8 binaries.
As such, end users are much more likely to have Java 8 installed in their systems.

I know that this is a niche, but maybe the blog post could have a mention to where one should download the most recent JDK and/or JRE?

tgodzik commented:
> There's one thing that bugs me about Scala dropping Java 8: It is now going to be more troublesome to distribute Scala applications to desktop users.

It's a good point. Maybe Scala Native or JS would be useful for those cases, but let's discuss that once I create the contributors thread before publishing the blogpost.

He-Pin commented:
Hi, as actually everyone I know will just jump from Java 8 to Java 21, and our Java 17 Jvm is not well maintained too, Java 21 maybe considerable here too.

mkurz commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,117 @@
+- [Spark 4](https://github.com/apache/spark/pull/43005#issuecomment-1731344040)
+  will require JDK 17
+- [Play Framework](https://github.com/playframework/playframework/issues/11078)
+  already dropped JDK 8 and will stop supporting JDK 11 in the next major
```
```suggestion
  already dropped JDK 8 and will [stop supporting JDK 11](https://github.com/playframework/playframework/issues/12179) in the next major
```

tgodzik replied:
Thanks!

He-Pin commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,117 @@
+**TLDR;**
+
+API used in lazy val implementation, will be removed in a future JDK thus one of
+the Scala 3 Next minors in 2025 Q4 and the next LTS will drop JDK 8 support.
```
This should not be enough trigger I think, If we switch to `AtomicReferenceFieldUpdater`, then problem is gone.

liach replied:
`AtomicLongFieldUpdater` is the right tool. The actual field holding the computed value is accessed plain; it's safe due to the release-acquire effects of the corresponding long bit map field, that it's written before bits are written, and read after the bits are read.

tgodzik replied:
We do try to list more reasons than just it. TLDR is just short version of that. And we can always discuss everything properly under the contributors thread

He-Pin commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,117 @@
+## Intro
+
+For the compiler’s second LTS version, the Scala 3 compiler team will be
+dropping support for JDK 8 and is considering which later JDK to use.
```
Scala 2.12.x and 2.13.x will continue to support Java 8 I think.

tgodzik replied:
That's for sure, we can't drop support for a patch version

He-Pin commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,117 @@
+them. Thus, the benefits of this change will be felt ecosystem-wide.
+
+If we take a look at the timeline we'll see that JDK 8 was first published on
+18th March 2014, which is already over 10 years ago. Technology and especially
```
Better to link to the eol https://www.oracle.com/java/technologies/java-se-support-roadmap.html

tgodzik replied:
Added

He-Pin commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,117 @@
+which might directly impact issues within your business applications.
+
+Some larger projects within JVM and Scala ecosystems have already dropped JDK 8
+support or even went straight to 17, these include:
```
Java 11's premier Support Until September 2023

sjrd commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,117 @@
+community feedback and our investigations. The current estimate for the next LTS
+is Q4 2025.
+
+One of the major challenges here will be making sure that we are still able to
+use libraries compiled with earlier JDK 8 compatible Scala 3 versions in the
+versions with a new implementation of lazy values, which is not reliant on
+sun.misc.Unsafe.
```
```suggestion
One major challenge will be to preserve backward compatibility.
As we change the encoding of `lazy val`s, we must still be able to use libraries compiled with earlier Scala 3 versions.
```

sjrd commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,117 @@
+
+## How does it affect me?
+
+If you are using Scala 3 on JDK 8 do let us know! However the current line of
+LTS under 3.3.x will be supported for at least another year after the release of
+the next LTS version, which should give you plenty of time to migrate.
```
```suggestion
The current line of LTS under 3.3.x will be supported for at least another year after the release of
the next LTS version. That should give you plenty of time to migrate.

After that, no more security fixes will be available for Scala 3 versions supporting JDK 8.

If you are using Scala 3 on JDK 8, do let us know!
```

sjrd commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,117 @@
+LTS under 3.3.x will be supported for at least another year after the release of
+the next LTS version, which should give you plenty of time to migrate.
+
+If it's not possible for you to switch, be sure to send us your feedback so that
+we can figure out a way to help you migrate.
```
```suggestion
If it is not possible for you to switch, be sure to send us your feedback so that
we can figure out a way to help you migrate.
```

sjrd commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,117 @@
+
+## Discussions and further reading
+
+You can track the current work related to lazy values under
+[this issue](https://github.com/scala/scala3/issues/9013) and discuss the topic
+under soon to be posted thread on the
+[Scala contributors forum](https://contributors.scala-lang.org/)
```
```suggestion
You can track the current work related to lazy values under
[this issue](https://github.com/scala/scala3/issues/9013).

Discussions on the topic of dropping JDK 8 support will soon be available on the
[Scala contributors forum](https://contributors.scala-lang.org/)
```

sjrd reviewed (commented):
A bunch of suggestions for the English grammar and phrasing. Mostly cutting long sentences into several smaller, active voice sentences.

tgodzik commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,117 @@
+- [Play Framework](https://github.com/playframework/playframework/issues/11078)
+  already dropped JDK 8 and will stop supporting JDK 11 in the next major
+  release
+- [Akka](https://github.com/akka/akka/pull/32127) dropped JDK 8
+- [Spring 6](https://spring.io/blog/2022/11/16/spring-framework-6-0-goes-ga)
+  will support JDK 17 and up
```
I wanted to make it a bit less repetitive :sweat_smile: But I understand that could be confusing

tgodzik commented:
> A bunch of suggestions for the English grammar and phrasing. Mostly cutting long sentences into several smaller, active voice sentences.

Thanks! That was very thorough!

SethTisue commented:
@tgodzik okay, I think we have enough Scala and Java projects in the projects lists. (I separated them into two lists.)

If the lists seem long, well.... I really think it's really good ammunition.

SethTisue commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,130 @@
+
+An API used in the `lazy val` implementation will be removed in a future JDK.
+Its replacement is only available on JDK 9+. Therefore, the Scala 3 Next minors
+in 2025 Q4 and the next LTS will drop JDK 8 support.
```
```suggestion
in 2025 Q4 and the next LTS will drop JDK 8 support (and perhaps 11 as well).
```

SethTisue replied:
btw I still think the TL;DR is too technical and it should just read:

> An API used in the `lazy val` implementation will be removed in a future JDK. Its replacement is only available on JDK 9+. Therefore, the Scala 3 Next minors in 2025 Q4 and the next LTS will drop JDK 8 support.

at which point it might as well just be merged with the "Intro" section rather than being separate.

I do not think the `lazy val` detail is worth foregrounding so strongly. If it was mainly about that, we wouldn't be considering dropping 11 too.

BalmungSan replied:
What happened to the consideration of going directly to `21`?

tgodzik replied:
> btw I still think the TL;DR is too technical and it should just read:

Yeah, that is probably better, changed.

SethTisue commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,130 @@
+---
+category: announcement
+permalink: /news/next-scala-lts.html
+title: "Next Scala 3 LTS version"
```
```suggestion
title: "Next Scala 3 LTS version will increase minimum required JDK version"
```

SethTisue commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,130 @@
+## Intro
+
+For the compiler’s second LTS version, the Scala 3 compiler team will be
+dropping support for JDK 8 and is considering which later JDK to use.
```
```suggestion
dropping support for JDK 8 and is considering which later JDK to require —
perhaps 11, perhaps 17.
```

SethTisue commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,130 @@
+## Immediate motivation
+
+The recent [JEP 471](https://openjdk.org/jeps/471) stated that the memory-access
+methods in sun.misc.Unsafe are scheduled for removal in a future release.
```
```suggestion
methods in `sun.misc.Unsafe` are scheduled for removal in a future release.
```

SethTisue commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,130 @@
+
+The recent [JEP 471](https://openjdk.org/jeps/471) stated that the memory-access
+methods in sun.misc.Unsafe are scheduled for removal in a future release.
+Currently, Scala 3 uses sun.misc.Unsafe in its implementation of lazy values.
```
```suggestion
Currently, Scala 3 uses `sun.misc.Unsafe` in its implementation of lazy values.
```

SethTisue commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,130 @@
+methods in sun.misc.Unsafe are scheduled for removal in a future release.
+Currently, Scala 3 uses sun.misc.Unsafe in its implementation of lazy values.
+This was needed due to compatibility with JDK 8. In order to support JDK 25+, we
+will need to drop usage of Unsafe. We are investigating the required changes
```
```suggestion
will need to drop usage of `Unsafe`. We are investigating the required changes
```

SethTisue commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,130 @@
+Currently, Scala 3 uses sun.misc.Unsafe in its implementation of lazy values.
+This was needed due to compatibility with JDK 8. In order to support JDK 25+, we
+will need to drop usage of Unsafe. We are investigating the required changes
+under [this issue](https://github.com/scala/scala3/issues/9013). It also started
```
insert paragraph break before "It"

SethTisue commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,130 @@
+This was needed due to compatibility with JDK 8. In order to support JDK 25+, we
+will need to drop usage of Unsafe. We are investigating the required changes
+under [this issue](https://github.com/scala/scala3/issues/9013). It also started
+to dawn on the compiler team that we might consider stopping to support JDK 8
```
```suggestion
to dawn on the compiler team that we might consider dropping support for JDK 8
```

SethTisue commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,130 @@
+
+Switching to a newer version of the JDK would allow both the compiler and the
+Scala ecosystem to start using new features and standard library improvements
+brought in by JDK 9+. brought in by JDK 9+. This might include a number of JEPs,
```
```suggestion
brought in by JDK 9+. This might include a number of JEPs (JDK Enhancement Proposals),
```

SethTisue commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,130 @@
+  the board
+
+It is always possible not to update your libraries and compiler version, which
+will avoid having to switch to a newer JDK. However, it is highly discouraged,
```
```suggestion
will avoid having to switch to a newer JDK. However, that is highly discouraged,
```

SethTisue commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,130 @@
+## Plan for making the switch
+
+The current plan is to drop support of JDK 8 in one of the future minor releases
+of Scala 3 and in the next LTS. The JDK will be either 11 or 17, depending on
```
```suggestion
of Scala 3 and in the next LTS. The new minimum JDK will be either 11 or 17, depending on
```

SethTisue commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,130 @@
+## Discussions and further reading
+
+You can track the current work related to lazy values under
+[this issue](https://github.com/scala/scala3/issues/9013). Discussions on the
```
insert paragraph break before "Discussions"

SethTisue commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,130 @@
+The current line of LTS under 3.3.x will be supported for at least another year
+after the release of the next LTS version. That should give you plenty of time
+to migrate. After that, no more security fixes will be available for Scala 3
+versions supporting JDK 8. If you are using Scala 3 on JDK 8, do let us know!
```
move the last sentence to the following paragraph

SethTisue reviewed (changes requested):
LGTM otherwise

lrytz reviewed (commented):
This looks good to me!

mkurz commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,130 @@
+Java examples include:
+
+- [Spring 6](https://spring.io/blog/2022/11/16/spring-framework-6-0-goes-ga)
+  will require JDK 17
```
```suggestion
  requires JDK 17
```

tgodzik commented:
I will post on contributors, should I essentially copy over the contents, so that people get the full picture?

SethTisue commented:
I don't see a need to copy-and-paste when you can just link to the blog post.

I'm doing a last copyedit now.

SethTisue reviewed (approved):
I pushed two commits with a number of small tweaks.

WojciechMazur commented on _posts/2025-02-01-next-scala-lts.md:
```diff
@@ -0,0 +1,125 @@
+---
+category: announcement
+permalink: /news/next-scala-lts.html
+title: "Next Scala 3 LTS version will increase minimum required JDK version"
```
Maybe we should use `LTS series` instead of `LTS version`. Otherwise just from headline somebody might think that Scala 3.3.6 LTS would require newer JDK.

tgodzik replied:
sure, updated!

tgodzik commented:
> I pushed two commits with a number of small tweaks.

And I removed them by mistake :sweat_smile: 

I added back the changes, sorry! I will update the table in the next PR
